heat/cluster/spectral.py

```diff
@@ -0,0 +1,309 @@
+import torch
+import math
+import numpy as np
+import heat as ht
+import time
+
+
+def laplacian(S, norm=True, mode="fc", upper=None, lower=None):
```

Comment:
this should probably be a hidden function, i.e. two leading underscores

Reply:
No. I put it as a stand alone function on purpose, since graph laplcians are not only relevant to spectral clustering, and could be used in other situations. However, API will be redesigned to make the function construct the graph laplacian directly from the dataset, i.e. include the similarity calculation. That way, it is more autonomnous.

Reply:
I see, agreed. Do you have an idea where this should go then from an API perspective? I am not sure that it is appropriate in the cluster module

Reply:
That is true. Phew, no idea. Linalg perhaps?

Reply:
How about ht.graph.lapclian

Reply:
so a new module for graphs? fine by me